plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/catalog/nessie/TestTrinoNessieCatalog.java

```diff
@@ -0,0 +1,152 @@
+import static io.trino.testing.sql.TestTable.randomTableSuffix;
+import static java.nio.file.Files.createTempDirectory;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static org.assertj.core.api.Assertions.fail;
+
+public class TestTrinoNessieCatalog
+        extends BaseTrinoCatalogTest
+{
+    private NessieContainer nessieContainer;
+
+    @BeforeClass
+    public void setupServer()
+    {
+        nessieContainer = NessieContainer.builder().build();
+        nessieContainer.start();
+    }
+
+    @AfterClass
+    public void teardownServer()
+    {
+        nessieContainer.close();
+    }
+
+    @Override
+    protected TrinoCatalog createTrinoCatalog(boolean useUniqueTableLocations)
+    {
+        Path tmpDirectory = null;
+        try {
+            tmpDirectory = createTempDirectory("test_nessie_catalog_warehouse_dir_");
+        }
+        catch (IOException e) {
+            fail(e.getMessage());
+        }
+        HdfsEnvironment hdfsEnvironment = new HdfsEnvironment(new HiveHdfsConfiguration(
+                new HdfsConfigurationInitializer(
+                        new HdfsConfig(),
+                        ImmutableSet.of()),
+                ImmutableSet.of()),
+                new HdfsConfig(),
+                new NoHdfsAuthentication());
+        NessieConfig nessieConfig = new NessieConfig();
+        nessieConfig.setServerUri(nessieContainer.getRestApiUri());
+        NessieApiV1 nessieApi = HttpClientBuilder.builder()
+                .withUri(nessieContainer.getRestApiUri())
+                .build(NessieApiV1.class);
+        NessieIcebergClient nessieClient = new NessieIcebergClient(nessieApi, nessieConfig.getDefaultReferenceName(), null);
+        return new TrinoNessieCatalog(
+                new CatalogName("catalog_name"),
+                new TestingTypeManager(),
+                new NessieIcebergTableOperationsProvider(new HdfsFileIoProvider(hdfsEnvironment), nessieClient),
+                nessieClient,
+                tmpDirectory.toAbsolutePath().toString(),
+                "testVersion",
+                useUniqueTableLocations);
+    }
+
+    @Test
+    public void testDefaultLocation()
+            throws IOException
+    {
+        Path tmpDirectory = createTempDirectory("test_nessie_catalog_default_location_");
+        tmpDirectory.toFile().deleteOnExit();
+        HdfsEnvironment hdfsEnvironment = new HdfsEnvironment(new HiveHdfsConfiguration(
+                new HdfsConfigurationInitializer(
+                        new HdfsConfig(),
+                        ImmutableSet.of()),
+                ImmutableSet.of()),
+                new HdfsConfig(),
+                new NoHdfsAuthentication());
+        NessieConfig nessieConfig = new NessieConfig();
+        nessieConfig.setDefaultWarehouseDir(tmpDirectory.toAbsolutePath().toString());
+        nessieConfig.setServerUri(nessieContainer.getRestApiUri());
```

Comment:
We could also try something like this, if required -
```
        NessieConfig nessieConfig = new NessieConfig()
                .setDefaultWarehouseDir(tmpDirectory.toAbsolutePath().toString())
                .setServerUri(nessieContainer.getRestApiUri());
```

Reply:
good point, updated